RecoTracker/TkSeedingLayers/interface/SeedingHitSet.h

```diff
@@ -11,40 +19,65 @@ class SeedingHitSet {
 
   static ConstRecHitPointer nullPtr() { return nullptr; }
 
-  SeedingHitSet() { theRecHits[0] = theRecHits[1] = theRecHits[2] = theRecHits[3] = nullptr; }
+  SeedingHitSet() = default;
 
-  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two)
-  // : theRecHits{{one,two,ConstRecHitPointer()}}
-  {
+  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two) {
+    initContainer();
     theRecHits[0] = one;
     theRecHits[1] = two;
-    theRecHits[2] = theRecHits[3] = nullptr;
+    setSize();
   }
-  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three)
-  // : theRecHits{{one,two,three}},
-  {
+
+  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three) {
+    initContainer();
     theRecHits[0] = one;
     theRecHits[1] = two;
     theRecHits[2] = three;
-    theRecHits[3] = nullptr;
+    setSize();
   }
 
   SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three, ConstRecHitPointer four) {
+    initContainer();
     theRecHits[0] = one;
     theRecHits[1] = two;
     theRecHits[2] = three;
     theRecHits[3] = four;
+    setSize();
+  }
+
+  SeedingHitSet(std::vector<ConstRecHitPointer> &hits, bool allowTruncation = false) {
+    initContainer();
+    auto minSize = std::min(hits.size(), size_t(seedinghits::kMaxSeedSize));
+
+    if (allowTruncation)
+      std::copy(hits.begin(), hits.begin() + minSize, theRecHits.begin());
+    else {
+      assert(hits.size() < size_t(seedinghits::kMaxSeedSize));
+      std::copy(hits.begin(), hits.begin() + minSize, theRecHits.begin());
+    }
+
+    setSize();
   }
 
-  ConstRecHitPointer const *data() const { return theRecHits; }
+  ConstRecHitPointer const *data() const { return theRecHits.data(); }
 
-  unsigned int size() const { return theRecHits[3] ? 4 : (theRecHits[2] ? 3 : (theRecHits[1] ? 2 : 0)); }
+  unsigned int size() const { return theSize; }
 
   ConstRecHitPointer get(unsigned int i) const { return theRecHits[i]; }
   ConstRecHitPointer operator[](unsigned int i) const { return theRecHits[i]; }
 
 private:
-  ConstRecHitPointer theRecHits[4];
+  std::array<ConstRecHitPointer, seedinghits::kMaxSeedSize> theRecHits;
```

Comment:
uhm, somehow I thought that the latest update was meant to use `std::vector`

Reply:
I have it [here](https://github.com/AdrianoDee/cmssw/commit/8739be727b173049782619339505551dd556b17e). Was trying to understand if I need to push it and "overwrite" the commits here. I'll do it now.